returnn/tf/network.py

```diff
@@ -1054,6 +1054,8 @@ def add_layer(self, name, layer_class, **layer_desc):
           layer = subnet.net.add_layer(name=sub_name, layer_class=layer_class, **layer_desc)
       else:
         layer = self._create_layer(name=name, layer_class=layer_class, **layer_desc)
+    if "output" in name and name in self.layers:
```

Comment:
This looks very heuristically. The check for some specific layer name.
This doesn't really make sense.
Why are layers with "output" in the name special?
I highly doubt that this is correct in general.

Reply:
I actually agree. My Idea was to check whether and what this breaks, to maybe get some more information about the error and the execution routine, because while I understand the steps that lead to this issue I am not really sure, where to actually look for correct handling.